distribution/lib/Standard/Test/0.0.0-dev/src/Bench.enso

```diff
@@ -1,7 +1,68 @@
 from Standard.Base import all
 import Standard.Base.Runtime.Ref.Ref
 
+
+type Bench_Options
+    ## PRIVATE
+    Impl iter_size num_iters
+
+    size : Integer -> Bench_Options
+    size self v = Bench_Options.Impl v self.num_iters
+
+    iter : Integer -> Bench_Options
+    iter self v = Bench_Options.Impl self.iter_size v
+
+    to_text self = "[iter_size=" + self.iter_size.to_text + ", num_iters=" + self.num_iters.to_text + "]"
+
+type Bench_Builder
+    ## PRIVATE
+    Impl builder
+
+    group : Text -> Bench_Options -> (Group_Builder -> Any) -> Any
+    group self (name:Text) (configuration:Bench_Options) fn =
+        b = Vector.new_builder
+        fn (Group_Builder.Impl b)
+        self.builder.append <| Bench.Group name configuration b.to_vector
+
+type Group_Builder
+    ## PRIVATE
+    Impl builder
+
+    specify : Text -> Any -> Bench
+    specify self (name:Text) ~benchmark =
+        self.builder.append <| Bench.Spec name (_ -> benchmark)
+
+
 type Bench
+    All (groups : Vector Bench)
+    Group (name:Text) (configuration:Bench_Options) (specs : Vector Bench)
```

Comment:
Is there a particular reason why configuration is bounded for a specific group? What about having configuration for `Spec`? I can imagine having three different Specs inside one group with different configs.

Reply:
No real justification other than I like it ;-)

Seriously: I am not sure what would be the purpose of `Group` otherwise? When it holds `configuration`, the group is a place to collect similar `specify` cases with similar configuration. 

>  three different Specs inside one group with different configs

_"Just create three different groups"_ would be my answer.

Seriously: the decision probably depends on _"what is more frequent?"_ If it is more common that multiple `specify` share the same config, then current design will work well.

Reply:
Sounds good. Let's keep the configuration per group then.